drivers/serial/uart_nrfx_uart.c

```diff
@@ -407,9 +414,12 @@ static int uart_nrfx_tx_abort(struct device *dev)
 static int uart_nrfx_rx_enable(struct device *dev, u8_t *buf, size_t len,
 			       s32_t timeout)
 {
+	__ASSERT(RX_PIN_USED(), "TX only UART instance");
```

Comment:
What about changing it to:
```
if (!RX_PIN_USED()) {
	__ASSERT(false, "TX only UART instance");
	return -ENOTSUP;
}
```
(and using similar code also for UARTE)? The rest of the function would be then optimized out when RX pin is not used.